src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1526,11 +1526,18 @@ def iter_patches(self, templated_file: TemplatedFile) -> Iterator[FixPatch]:
         else:
             # This segment isn't a literal, but has changed, we need to go deeper.
 
+            # If there's an end of file segment or indent, ignore it just for the
+            # purposes of patch iteration.
+            # NOTE: This doesn't mutate the underlying `self.segments`.
+            segments = self.segments
+            if self.segments[-1].is_type("end_of_file", "indent"):
+                segments = segments[:-1]
```

Comment:
I suspect the answer is "no", but is it possible (and do we care) if the last *two* segments in the list are:
* indent
* end_of_file

Reply:
hmmmm good question. It's not happened yet, but I guess in theory it probably is. I'll see if I can account for that.

Reply:
Shameless sales pitch: IIRC, the Segments class in the functional API has a `last()` function which makes this easy.

Reply:
Good idea 👍

Reply:
Hmmm - actually from here in the codebase I'd rather not add more external dependents. If this was in the reflow module I would totally be with you.